File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        count = 0
        stack = [(0, 0)]
        while stack:
            cur_sum, idx = stack.pop()
            if idx == len(nums):
                if cur_sum == target:
                    count += 1
                continue

            stack.append((cur_sum - nums[idx], idx + 1))
            stack.append((cur_sum + nums[idx], idx + 1))
        return count
        

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> d;
        for (int i=0; i<nums.size(); i++) {
            int n = nums[i];
            if (d.count(n) && abs(d[n] - i) <= k) {
                return true;
            }
            d[n] = i;
        }
        return false;
        
    }
};